### PR TITLE
Parsing compostional keywords

### PR DESCRIPTION
**Pull request by GitPaean**
So far, it includes the parsing of the following keyword:
COMPS, EOS, ACF, MW, CNAMES, PCRIT, TCRIT, VCRIT, BIC and EOSNUM. 

EOSNUM likely needs some more checking when we need it. 

Did not implement the XMF and YMF (SOLUTION section), which look like need some special care and they will likely to be in a separate PR.

### CODE REVIEW

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
It is the beginning of parsing compositional keywords. We do not have much experience about the syntax side of these keywords, which keywords are mandatory, whether keywords can overwrite the previous ones, and so on.  And also, we do not know how we will use the information from the simulator side.  So it will evolve when we develop the downstream parts and also with more testing. 

This PR provides one way to parse some compositional keywords, which is a start.  There will be some other keywords need to be parsed before we do a simulation, they will be addressed in separate PRs. 

I am marking this PR ready for review. So the reviewing and discussion can start.

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
It looks like this PR affects some CO2 related DATA file we have but I suddenly could not see the jenkins report. 

Will trigger a new one.

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by bska** [comment]
> I suddenly could not see the jenkins report.

Right.  There's been quite a few OPM-Common build checks in the past few hours and the CI system only retains the 10 most recent builds.

> Will trigger a new one.

That's the only course of action in this case.

**Comment by GitPaean** [comment]
Will check why this PR affects the following two cases, 

[Test Result](https://ci.opm-project.org/job/opm-common-PR-builder/6759/testReport/) (2 failures / +2)
[mpi.compareECLFiles_flow+CO2STORE_GASWAT](https://ci.opm-project.org/job/opm-common-PR-builder/6759/testReport/junit/(root)/mpi/compareECLFiles_flow_CO2STORE_GASWAT/)
[mpi.compareECLFiles_flow+H2STORE_GASWAT](https://ci.opm-project.org/job/opm-common-PR-builder/6759/testReport/junit/(root)/mpi/compareECLFiles_flow_H2STORE_GASWAT/)

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

File: opm/input/eclipse/EclipseState/Grid/FieldProps.cpp [hunk 1]
```diff
@@ -45,6 +45,8 @@
 #include <opm/input/eclipse/Parser/ParserKeywords/O.hpp>
 #include <opm/input/eclipse/Parser/ParserKeywords/P.hpp>
 #include <opm/input/eclipse/Parser/ParserKeywords/T.hpp>
+#include <opm/input/eclipse/Parser/ParserKeywords/X.hpp>
+#include <opm/input/eclipse/Parser/ParserKeywords/Y.hpp>
```
Comment (totto82) [inline opm/input/eclipse/EclipseState/Grid/FieldProps.cpp:50]: Why is this needed?

**Review by totto82 — APPROVED** [review]
Thanks. I only have one question related to num_eos_res. You read the number from TABDIMS, but that is all. For now I think that is ok. I guess you can add the related keywords later.

**Comment by GitPaean** [comment]
> I only have one question related to num_eos_res. You read the number from TABDIMS, but that is all.

Yes. And many keywords using it for the sizing. Can you elaborate more about what you are trying to say here?

image

File: opm/input/eclipse/EclipseState/Grid/FieldProps.cpp (hunk 1 above)
Comment (GitPaean) [reply opm/input/eclipse/EclipseState/Grid/FieldProps.cpp:50]: They are needed for XMF and YMF.  Should not be in this PR.

**Comment by GitPaean** [comment]
> > I only have one question related to num_eos_res. You read the number from TABDIMS, but that is all.
> 
> Yes. And many keywords using it for the sizing. Can you elaborate more about what you are trying to say here?
> 


I can see now. You are referring to `num_eos_sur`.  We do not many keywords need that number for now indeed.

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
With the approval, and jenkins approves. I am getting the PR in now.

**Comment by GitPaean** [comment]
It is something very new, and new development/changes along the line will be expected, but it is relatively isolated and easy to avoid to affect the running of the main development.
